[AOTI][refactor] Separate header codegen

desertfire opened the pull request:
Summary: Move arrayref specific header codegen logic to cpp_wrapper_cpu_array_ref.py, and consolidate some header files codegen logic

Test Plan: CI

Differential Revision: D64899248




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @chauhang @aakhundov

hl475 reviewed (approved):
I checked one of the failure from CI

> 2024-10-25T18:46:27.3082226Z ModuleNotFoundError: No module named 'torch._C._distributed_c10d'; 'torch._C' is not a package

I guess it's not relevant with this change?

desertfire committed:
[AOTI][refactor] Separate header codegen (#138882)

Summary:

Move arrayref specific header codegen logic to cpp_wrapper_cpu_array_ref.py, and consolidate some header files codegen logic

Test Plan: CI

Reviewed By: hl475

Differential Revision: D64899248
